Documentation/tutorial/howto_create_custom_template.md

```diff
@@ -1,44 +1,80 @@
-_locales | Array in JSON format. The list of locales supported for current page.
-_ext | The extension of the generated page
-_lang | The programming language of the page
-_name | The name of the page
-_description | The description of the page
-_rel | The relative path of the folder of the generated page to the root folder of the generated document
-_navPath | The relative path of the root TOC file, if exists
-_navRel | The relative path of the folder of the root TOC file to the root folder of the generated document, if exists
-_tocPath | The relative path of the TOC file, if exists
-_tocRel | The relative path of the folder of the TOC file to the root folder of the generated document, if exists
-
-### 2. What is the extension of the output file?
-The extension of the output file is defined by the naming of the template file `<type>[.<extension>].tmpl`. For example, with a template naming `markdown.md.tmpl`, the extension will be `.md`; with a template naming `markdown.tmpl`, the extension will be ``.
-
-### 3. How to transform the model before applying it to the template?
-If a `default.js` file exists in the same folder as the template file, this javascript file will be evalated. `docfx` tries to evaluate the javascript file and call `transform` function afterwards. `docfx` considers `transform` function to be in the following syntax:
+
+Templates are organized as a zip package or a folder. The zipped package file path without extension or the folder path is considered as the template name.
+
+Quick Start
+---------------
+Let's create a template to transform Markdown files into a simple html file.
+
+### Step 1. Create a template folder
+Create a folder for the template, for example, `c:/docfx_howto/simple_template`.
+
+### Step 2. Add *Renderer File*
+Create a file `conceptual.html.primary.tmpl` under the template folder, with the following content:
+
+```mustache
+{{{conceptual}}}
+```
+
+Now a simple custom template is created.
+
+You may notice that DocFX reports a warning message saying that: *Warning: [Build Document.Apply Templates]There is no template processing document type(s): Toc*. It is because our custom template only specifies how to handle document with type `conceptual`.
+
+To test the output of the template, create a simple documentation project following [Walkthrough Part I](walkthrough/walkthrough_create_a_docfx_project.md) or download the [zipped documentation project](walkthrough/artifacts/walkthrough1.zip) directly.
+
+In the documentation project, run `docfx build docfx.json -t c:/docfx_howto/simple_template --serve`. `-t` command option specifies the template name(s) used by current build.
+
+Open http://localhost:8080 and you can see a simple web page as follows:
+
+![Simple Web Page](images/simple_web_page.png)
+
+Add *Preprocessor File*
+-----------------------
+### Step 3. Add *Preprocessor File*
+You can modify the input data model before applying *Renderer File* with a *Preprocessor File*.
```

Comment:
Add a paragraph to describe what you're trying to achieve by adding preprocessor file.